### PR TITLE
Port from pbf->protozero@1.3.0

### PR DESCRIPTION
**Pull request by springmeyer**
Ports the codebase from pbf.hpp (a precursor to protozero) to protozero@1.3.0.

Incidentally fixes several `-fsanitize=integer` errors:

```
../src/mbgl/util/pbf.hpp:114:23: runtime error: negation of 1 cannot be represented in type 'unsigned int'
../src/mbgl/tile/vector_tile.cpp:124:15: runtime error: unsigned integer overflow: 4129 + 4294967234 cannot be represented in type 'unsigned int'
../src/mbgl/tile/vector_tile.cpp:125:15: runtime error: unsigned integer overflow: 4160 + 4294967235 cannot be represented in type 'unsigned int'
../src/mbgl/tile/vector_tile.cpp:125:15: runtime error: unsigned integer overflow: 4096 + 4294966706 cannot be represented in type 'unsigned int'
../src/mbgl/tile/vector_tile.cpp:124:15: runtime error: unsigned integer overflow: 4294967168 + 128 cannot be represented in type 'unsigned int'

```

### CODE REVIEW

File: src/mbgl/text/glyph_pbf.cpp [hunk 1]
```diff
@@ -6,52 +6,55 @@
...
-        if (glyphs_pbf.tag == 1) { // stacks
-            mbgl::pbf fontstack_pbf = glyphs_pbf.message();
-            while (fontstack_pbf.next()) {
-                if (fontstack_pbf.tag == 3) { // glyphs
-                    mbgl::pbf glyph_pbf = fontstack_pbf.message();
+    protozero::pbf_reader glyphs_pbf(data);
 
-                    mbgl::SDFGlyph glyph;
+    while (glyphs_pbf.next(1)) {
+        protozero::pbf_reader fontstack_pbf = glyphs_pbf.get_message();
```
Comment (kkaefer) [inline src/mbgl/text/glyph_pbf.cpp]: Can we use `auto` here?

File: src/mbgl/text/glyph_pbf.cpp [hunk 2]
```diff
@@ -6,52 +6,55 @@
...
-                        }
-                    }
+            mbgl::SDFGlyph glyph;
 
-                    glyphSet.insert(glyph.id, glyph);
-                } else {
-                    fontstack_pbf.skip();
+            while (glyph_pbf.next()) {
+                switch (glyph_pbf.tag())
+                {
```
Comment (kkaefer) [inline src/mbgl/text/glyph_pbf.cpp]: Our code style mandates opening braces to be on the same line as the control statement.

File: src/mbgl/tile/vector_tile.cpp [hunk 3]
```diff
@@ -59,19 +59,19 @@ optional<Value> VectorTileFeature::getValue(const std::string& key) const {
         return optional<Value>();
     }
 
-    pbf tags = tags_pbf;
-    while (tags) {
-        uint32_t tag_key = tags.varint();
+    auto tags = tags_iter;
```
Comment (kkaefer) [inline src/mbgl/tile/vector_tile.cpp]: Can we just copy `tags_iter.first`, since `tags.second` will never be modified?

Comment (springmeyer) [inline src/mbgl/tile/vector_tile.cpp]: Good idea @kkaefer - done in 9f1ade4

File: src/mbgl/text/glyph_pbf.cpp (hunk 2 above)
Comment (springmeyer) [inline src/mbgl/text/glyph_pbf.cpp]: Thanks for catching that, done in 9f1ade4

File: src/mbgl/text/glyph_pbf.cpp (hunk 1 above)
Comment (springmeyer) [inline src/mbgl/text/glyph_pbf.cpp]: Good idea, done in 9f1ade4

**Comment by springmeyer** [comment]
Ready for final 👀 before merging.

**Comment by jfirebaugh** [comment]
👍

File: src/mbgl/tile/vector_tile.cpp [hunk 4]
```diff
@@ -86,10 +92,11 @@ optional<Value> VectorTileFeature::getValue(const std::string& key) const {
...
 std::unordered_map<std::string,Value> VectorTileFeature::getProperties() const {
     std::unordered_map<std::string,Value> properties;
-    pbf tags = tags_pbf;
-    while (tags) {
-        uint32_t tag_key = tags.varint();
-        uint32_t tag_val = tags.varint();
+    auto start_itr = tags_iter.first;
+    const auto & end_itr = tags_iter.second;
+    while (start_itr != end_itr) {
+        uint32_t tag_key = static_cast<uint32_t>(*start_itr++);
```
Comment (joto) [inline src/mbgl/tile/vector_tile.cpp]: No check here for `start_itr == end_itr` like in line 76.

Comment (springmeyer) [inline src/mbgl/tile/vector_tile.cpp]: Thank you @joto - fixed in 0c6bce2

**Pull request merged**